### PR TITLE
tree-wide: add missing magic checks

### PR DESCRIPTION
**Pull request by asadsa92**
Signed-off-by: Asad Sajjad Ahmed <asadsa@varnish-software.com>

### CODE REVIEW

File: lib/libvarnish/vre.c [hunk 1]
```diff
@@ -294,10 +294,9 @@ VRE_sub(const vre_t *code, const char *subject, const char *replacement,
 void
 VRE_free(vre_t **vv)
 {
-	vre_t *v = *vv;
+	vre_t *v;
 
-	*vv = NULL;
-	CHECK_OBJ(v, VRE_MAGIC);
+	TAKE_OBJ_NOTNULL(v, vv, VRE_MAGIC);
```
Comment (Dridi) [inline lib/libvarnish/vre.c:299]: How did I not see this one?

File: lib/libvarnishapi/vsc.c [hunk 2]
```diff
@@ -319,7 +319,6 @@ vsc_del_seg(const struct vsc *vsc, struct vsm *vsm, struct vsc_seg **spp)
 	struct vsc_seg *sp;
 
 	CHECK_OBJ_NOTNULL(vsc, VSC_MAGIC);
-	AN(vsm);
```
Comment (Dridi) [inline lib/libvarnishapi/vsc.c:322]: Why remove this one?

File: lib/libvarnishapi/vsc.c [hunk 3]
```diff
@@ -343,7 +342,6 @@ vsc_add_seg(const struct vsc *vsc, struct vsm *vsm, const struct vsm_fantom *fp)
 	struct vsc_pt *pp;
 
 	CHECK_OBJ_NOTNULL(vsc, VSC_MAGIC);
-	AN(vsm);
```
Comment (Dridi) [inline lib/libvarnishapi/vsc.c:346]: Why remove this one?

File: lib/libvarnishapi/vut.c [hunk 4]
```diff
@@ -378,7 +378,6 @@ VUT_Fini(struct VUT **vutp)
 	if (vut->vsm)
 		VSM_Destroy(&vut->vsm);
 
-	memset(vut, 0, sizeof *vut);
```
Comment (Dridi) [inline lib/libvarnishapi/vut.c:381]: Why remove this one? I know it has a high chance of being optimized out, but why remove it?

File: lib/libvarnishapi/vsc.c (hunk 2 above)
Comment (asadsa92) [reply lib/libvarnishapi/vsc.c:322]: We only pass the "object" so strictly speaking we do not have to check whether it satisfy our need.
Anyways, this should have been `CHECK_OBJ_NOTNULL(vsm, VSM_MAGIC);`, but that is already done by both `VSM_Map`/"VSM_Unmap`.

**Pull request merged**